ts/index.ts

```diff
@@ -0,0 +1,188 @@
+import * as actionsCore from "@actions/core";
+import * as actionsExec from "@actions/exec";
+import { ActionOptions, IdsToolbox, inputs } from "detsys-ts";
+
+const EVENT_EXECUTION_FAILURE = "execution_failure";
+
+const VISIBILITY_OPTIONS = ["public", "unlisted", "private"];
+
+type Visibility = "public" | "unlisted" | "private";
+
+type ExecutionEnvironment = {
+  FLAKEHUB_PUSH_VISIBILITY?: string;
+  FLAKEHUB_PUSH_TAG?: string;
+  FLAKEHUB_PUSH_HOST?: string;
+  FLAKEHUB_PUSH_LOG_DIRECTIVES?: string;
+  FLAKEHUB_PUSH_LOGGER?: string;
+  FLAKEHUB_PUSH_GITHUB_TOKEN?: string;
+  FLAKEHUB_PUSH_NAME?: string;
+  FLAKEHUB_PUSH_REPOSITORY?: string;
+  FLAKEHUB_PUSH_DIRECTORY?: string;
+  FLAKEHUB_PUSH_GIT_ROOT?: string;
+  FLAKEHUB_PUSH_EXTRA_LABELS?: string;
+  FLAKEHUB_PUSH_SPDX_EXPRESSION?: string;
+  FLAKEHUB_PUSH_ERROR_ON_CONFLICT?: string;
+  FLAKEHUB_PUSH_INCLUDE_OUTPUT_PATHS?: string;
+  FLAKEHUB_PUSH_ROLLING?: string;
+  FLAKEHUB_PUSH_MIRROR?: string;
+  FLAKEHUB_PUSH_ROLLING_MINOR?: string;
+};
+
+class FlakeHubPushAction {
+  idslib: IdsToolbox;
+
+  // Action inputs translated into environment variables to pass to flakehub-push
+  private tag: string;
+  private host: string;
+  private logDirectives: string;
+  private logger: string;
+  private gitHubToken: string;
+  private repository: string;
+  private directory: string;
+  private gitRoot: string;
+  private extraLabels: string;
+  private spdxExpression: string;
+  private errorOnConflict: boolean;
+  private includeOutputPaths: boolean;
+  private rolling: boolean;
+  private mirror: boolean;
+  private name: string | null;
+  private rollingMinor: number | null;
+
+  constructor() {
+    const options: ActionOptions = {
+      name: "flakehub-push",
+      fetchStyle: "gh-env-style",
+      diagnosticsUrl: new URL(
+        "https://install.determinate.systems/flakehub-push/telemetry",
+      ),
+      legacySourcePrefix: "flakehub-push",
+      requireNix: "fail",
+    };
+
+    this.idslib = new IdsToolbox(options);
+
+    // Inputs translated into environment variables for flakehub-push
+    this.tag = inputs.getString("tag");
+    this.host = inputs.getString("host");
+    this.logDirectives = inputs.getString("log-directives");
+    this.logger = inputs.getString("logger");
+    this.gitHubToken = inputs.getString("github-token");
+    this.repository = inputs.getString("repository");
+    this.directory = inputs.getString("directory");
+    this.gitRoot = inputs.getString("git-root");
+    // extra-tags is deprecated but we still honor it
+    this.extraLabels =
+      inputs.getString("extra-labels") === ""
+        ? inputs.getString("extra-tags")
+        : "";
```

Comment:
This looks wrong. Seems to be saying "if extra labels is empty, use the value from extra tags; otherwise, use the empty string".

Reply:
As far as I can tell, that's the behavior we want here. `extra-labels` is the up-to-date name of the input while `extra-tags` is the backup. If neither is set, then empty string is valid, as the CLI expects a comma-separated list.

Reply:
Ah okay, I see the issue